lib/utils.bash

```diff
@@ -111,7 +111,9 @@ check_if_plugin_exists() {
     exit 1
   fi
 
-  if [ ! -d "$(asdf_data_dir)/plugins/$plugin_name" ]; then
+  # ".." is evaluated to the parent plugins folder, which should not be valid
+  if [ ".." == "${plugin_name}" ] \
```

Comment:
Were you actually seeing this function invoked with a plugin name of `..`? If so this a good clue.

Reply:
I'm still not sure how exactly I managed to get `..` into the metadata, so this is more of a conjecture after looking over the code for a while.

At the very least, without this check, I'm pretty sure that if `..` comes as an argument to `check_if_plugin_exist`, it would be recognized as valid, since `..` always exists in the plugin folder (even when empty).

I left this in a separate commit in case the function I made was rejected, since I thought it's worthwhile to investigate this further.

Reply:
That's a good observation. I am going to do some testing locally to try to figure out what code is invoking this function with `..`, I tested it with the `reshim` command and it doesn't appear to ever pass in `..` for me.